### PR TITLE
Update minimap2 to 2.12

### PR DESCRIPTION
**Pull request by mvdbeek**
FOR CONTRIBUTOR:
* [ ] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [ ] - License permits unrestricted use (educational + commercial)
* [ ] - This PR adds a new tool or tool collection
* [x] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

FOR REVIEWER:
* [ ] .shed.yml file ok
    - [ ] Toolshed user `iuc` has access to associated toolshed repo(s)
* [ ] Indentation is correct (4 spaces)
* [ ] Tool version/build ok
* [ ] `<command/>`
  - [ ] Text parameters, input and output files `'single quoted'`
  - [ ] Use of `<![CDATA[ ... ]]>` tags
  - [ ] Parameters of type `text` or having `optional="true"` attribute are checked with `if str($param)` before being used
* [ ] Data parameters have a `format` attribute containing datatypes recognised by Galaxy
* [ ] Tests
  - [ ] Parameters are reasonably covered
  - [ ] Test files are appropriate
* [ ] Help
  - [ ] Valid restructuredText and uses `<![CDATA[ ... ]]>` tags
* [ ] Complies with other best practice in [Best Practices Doc](http://galaxy-iuc-standards.readthedocs.io/en/latest/best_practices/tool_xml.html)

### CODE REVIEW

File: tools/minimap2/minimap2.xml [hunk 1]
```diff
@@ -2,8 +2,8 @@
 <tool id="minimap2" name="Map with minimap2" version="2.5+gx1" profile="17.01">
```
Comment (nsoranzo) [inline tools/minimap2/minimap2.xml]: Can you update the tool version? I'd add a `@TOOL_VERSION@` macro token to be used here and in the `<requirement />`.

File: tools/minimap2/minimap2.xml [hunk 2]
```diff
@@ -171,17 +173,19 @@
...
+            <option value="sr">Short single-end reads without splicing (-k21 -w11 --sr --frag=yes -A2 -B8 -O12,32 -E2,1 -r50 -p.5 -N20 -f1000,5000 -n2 -m20 -s40 -g200 -2K50m --heap-sort=yes --secondary=no)</option>
         </param>
-        <section name="mapping_options" title="Set advanced mapping options" help="Sets -f, -g, -G, -F, -r, -n, -m, -X, -p and -N options." expanded="False">
+        <section name="mapping_options" title="Set advanced mapping options" help="Sets -f, -g, -G, -F, -r, -n, -m, -X, -p, -N and --min-occ-floor options." expanded="False">
             <param argument="-f" type="float" value="" optional="true" label="filter out top FLOAT fraction of repetitive minimizers" help="default=0.0002"/>
+            <param argument="--min-occ-floor" name="min_occ_floor" type="integer" label="force minimap2 to always use k-mers occuring this many times or fewer" help="Maximum occurence is the number of repetitive minimizers determined by'-f' or this value, whichever is higher." optional="true" />
```
Comment (nsoranzo) [inline tools/minimap2/minimap2.xml]: `determined by'-f' or`: missing space after `by`

File: tools/minimap2/minimap2.xml [hunk 3]
```diff
@@ -221,6 +225,7 @@
                 <option value="short">short</option>
                 <option value="long">long</option>
             </param>
+            <param argument="--eqx" type="boolean" truevalue="--eqx" falsevalue="" label="write =/X CIGAR operators"/>
```
Comment (nsoranzo) [inline tools/minimap2/minimap2.xml]: This seems to be unused in the `<command />`.

**Comment by mvdbeek** [comment]
Thanks for the review @nsoranzo ! Can't believe I missed these things ...

**Comment by nsoranzo** [comment]
Should this be rebased now?

**Comment by mvdbeek** [comment]
Yep, just done that

**Pull request merged**